metrics/web/test/dashboard/presentation/widgets/build_result_bar_component_test.dart

```diff
@@ -0,0 +1,423 @@
+
+import 'package:flutter/gestures.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:metrics/base/presentation/graphs/placeholder_bar.dart';
+import 'package:metrics/common/presentation/graph_indicator/widgets/metrics_graph_indicator.dart';
+import 'package:metrics/common/presentation/metrics_theme/config/dimensions_config.dart';
+import 'package:metrics/common/presentation/metrics_theme/model/metrics_theme_data.dart';
+import 'package:metrics/common/presentation/metrics_theme/model/metrics_widget_theme_data.dart';
+import 'package:metrics/dashboard/presentation/view_models/build_result_popup_view_model.dart';
+import 'package:metrics/dashboard/presentation/view_models/build_result_view_model.dart';
+import 'package:metrics/dashboard/presentation/view_models/finished_build_result_view_model.dart';
+import 'package:metrics/dashboard/presentation/view_models/in_progress_build_result_view_model.dart';
+import 'package:metrics/dashboard/presentation/widgets/build_result_bar.dart';
+import 'package:metrics/dashboard/presentation/widgets/build_result_bar_component.dart';
+import 'package:metrics/dashboard/presentation/widgets/build_result_popup_card.dart';
+import 'package:metrics/dashboard/presentation/widgets/strategy/build_result_bar_padding_strategy.dart';
+import 'package:metrics/dashboard/presentation/widgets/strategy/build_status_graph_indicator_appearance_strategy.dart';
+import 'package:metrics_core/metrics_core.dart';
+import 'package:mockito/mockito.dart';
+import 'package:network_image_mock/network_image_mock.dart';
+import 'package:plugin_platform_interface/plugin_platform_interface.dart';
+import 'package:url_launcher_platform_interface/url_launcher_platform_interface.dart';
+
+import '../../../test_utils/matchers.dart';
+import '../../../test_utils/metrics_themed_testbed.dart';
+
+// ignore_for_file: avoid_redundant_argument_values
+
+void main() {
+  group("BuildResultBarComponent", () {
+    const metricsTheme = MetricsThemeData(
+      inactiveWidgetTheme: MetricsWidgetThemeData(
+        primaryColor: Colors.red,
+      ),
+    );
+
+    final placeholderFinder = find.byType(PlaceholderBar);
+    final graphIndicatorFinder = find.byWidgetPredicate(
+      (widget) {
+        return widget is MetricsGraphIndicator<BuildStatus>;
+      },
+    );
+
+    final mouseRegionFinder = find.ancestor(
+      of: find.byType(GestureDetector),
+      matching: find.byType(MouseRegion),
+    );
+    final stackFinder = find.ancestor(
+      of: find.byType(BuildResultBar),
+      matching: find.byType(Stack),
+    );
+
+    Future<void> hoverBar(WidgetTester tester) async {
+      final mouseRegion = tester.firstWidget<MouseRegion>(mouseRegionFinder);
+      mouseRegion.onEnter(const PointerEnterEvent());
+
+      await mockNetworkImagesFor(() {
+        return tester.pumpAndSettle();
+      });
+    }
+
+    final popupViewModel = BuildResultPopupViewModel(
```

Comment:
Would it be great to specify this field before functions declaration? The same suggestion for the `urlLauncherMock` field.